### PR TITLE
feat(uat): add and remove steps for loopback address

### PR DESCRIPTION
**Pull request by auarbekov**
**Issue #, if available:**
https://klika-tech.atlassian.net/browse/GGMQ-203
add and remove steps for loopback address

**Description of changes:**
- Add add loopback address steps
- Add remove loopback address steps

**Why is this change necessary:**
Implement scenarios

**How was this change tested:**
Run scenario on CodeBuild

**Test results:**
```
Given my device is registered as a Thing....................................passed
And my device is running Greengrass.........................................passed
When I create a Greengrass deployment with components.......................passed
And I create client device "basic_connect"..................................passed
When I associate "basic_connect" with ggc...................................passed
And I update my Greengrass deployment configuration, setting the component aws.greengrass.clientdevices.Auth configuration to:.passed
And I update my Greengrass deployment configuration, setting the component aws.greengrass.client.Mqtt5JavaSdkClient configuration to:.passed
And I update my Greengrass deployment configuration, setting the component aws.greengrass.clientdevices.IPDetector configuration to:.passed
And I deploy the Greengrass deployment configuration........................passed
Then the Greengrass deployment is COMPLETED on the device after 5 minutes...passed
And the aws.greengrass.clientdevices.mqtt.EMQX log on the device contains the line "is running now!." within 1 minutes.passed
Then I add loop back address "127.0.0.2"....................................passed
Then I wait 90 seconds......................................................passed
And I discover core device broker as "default_broker" from "basic_connect" in OTF.passed
And I connect device "basic_connect" on aws.greengrass.client.Mqtt5JavaSdkClient to "default_broker" using mqtt "v5".passed
And I disconnect device "basic_connect" with reason code 0..................passed
Then I remove loop back address "127.0.0.2".................................passed
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java [hunk 1]
```diff
@@ -77,6 +78,32 @@ public void setDeviceMqtt(final boolean connectivity) throws IOException, Interr
         }
     }
 
+    /**
+     * Add loop back address.
```
Comment (bgklika) [inline aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java]:
```suggestion
     * Add IP address to loopback interface
```

File: aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java [hunk 2]
```diff
@@ -77,6 +78,32 @@ public void setDeviceMqtt(final boolean connectivity) throws IOException, Interr
...
+     *
+     * @param address the value of loop back address
+     * @throws IOException on IO errors
+     * @throws InterruptedException when thread has been interrupted
+     */
+    @Then("I add loop back address {string}")
```
Comment (bgklika) [inline aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java]:
```suggestion
     I add IP address {string} to loop back interface
```

File: aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java [hunk 3]
```diff
@@ -77,6 +78,32 @@ public void setDeviceMqtt(final boolean connectivity) throws IOException, Interr
...
+    public void addLoopBackAddress(final String address) throws IOException, InterruptedException {
+        LOGGER.info("Adding loopback address {}", address);
+        platform.networkUtils().addLoopbackAddress(address);
+    }
+
+    /**
+     * Delete loop back address.
```
Comment (bgklika) [inline aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java]:
```suggestion
     I remove IP address {string} from loop back interface
```

File: aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java [hunk 4]
```diff
@@ -77,6 +78,32 @@ public void setDeviceMqtt(final boolean connectivity) throws IOException, Interr
...
+     *
+     * @param address the value of loop back address
+     * @throws IOException on IO errors
+     * @throws InterruptedException when thread has been interrupted
+     */
+    @Then("I remove loop back address {string}")
```
Comment (bgklika) [inline aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java]:
```suggestion
     I remove IP address {string} from loop back interface
```

File: aws-greengrass-testing-platform/aws-greengrass-testing-platform-api/src/main/java/com/aws/greengrass/testing/platform/linux/LinuxNetworkUtils.java [hunk 5]
```diff
@@ -51,6 +52,31 @@ public void recoverMqtt() throws InterruptedException, IOException {
         modifyMqttConnection(DISABLE_OPTION);
     }
 
+    @Override
```
Comment (bgklika) [inline aws-greengrass-testing-platform/aws-greengrass-testing-platform-api/src/main/java/com/aws/greengrass/testing/platform/linux/LinuxNetworkUtils.java:59]: Can simplify to move common code to add_or_delete method. Delete differs from add only in one parameter to the command.

Comment (auarbekov) [reply aws-greengrass-testing-platform/aws-greengrass-testing-platform-api/src/main/java/com/aws/greengrass/testing/platform/linux/LinuxNetworkUtils.java:59]: I think we can extend the method in the future and according **Single Responsibility** it's wrong.

File: aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java (hunk 1 above)
Comment (auarbekov) [reply aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java]: fixed in b0f3f75a79ffe49f7ffbb15e0ea553cc9deb6a7a

File: aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java (hunk 2 above)
Comment (auarbekov) [reply aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java]: fixed in b0f3f75a79ffe49f7ffbb15e0ea553cc9deb6a7a

File: aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java (hunk 3 above)
Comment (auarbekov) [reply aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java]: fixed in b0f3f75a79ffe49f7ffbb15e0ea553cc9deb6a7a

File: aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java (hunk 4 above)
Comment (auarbekov) [reply aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/NetworkUtilsSteps.java]: fixed in b0f3f75a79ffe49f7ffbb15e0ea553cc9deb6a7a

**Review by eschastlivtsev — APPROVED** [review]

**Review by bgklika — APPROVED** [review]

**Pull request closed without merging**